seatunnel-connectors-v2/connector-file/connector-file-base-hadoop/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hdfs/source/BaseHdfsFileSource.java

```diff
@@ -49,12 +51,26 @@ public void prepare(Config pluginConfig) throws PrepareFailException {
             throw new PrepareFailException(getPluginName(), PluginType.SOURCE, "Check file path fail.");
         }
         // support user-defined schema
+        FileFormat fileFormat = FileFormat.valueOf(pluginConfig.getString(HdfsSourceConfig.FILE_TYPE).toUpperCase());
+        // only json text csv type support user-defined schema now
         if (pluginConfig.hasPath(HdfsSourceConfig.SCHEMA)) {
-            Config schemaConfig = pluginConfig.getConfig(HdfsSourceConfig.SCHEMA);
-            rowType = SeaTunnelSchema
-                .buildWithConfig(schemaConfig)
-                .getSeaTunnelRowType();
-            readStrategy.setSeaTunnelRowTypeInfo(rowType);
+            switch (fileFormat) {
+                case CSV:
+                case TEXT:
+                case JSON:
+                    Config schemaConfig = pluginConfig.getConfig(SeaTunnelSchema.SCHEMA);
+                    rowType = SeaTunnelSchema
+                            .buildWithConfig(schemaConfig)
+                            .getSeaTunnelRowType();
+                    readStrategy.setSeaTunnelRowTypeInfo(rowType);
+                    break;
+                case ORC:
+                case PARQUET:
+                    throw new UnsupportedOperationException("SeaTunnel does not support user-defined schema for [parquet, orc] files");
```

Comment:
For Hive Source Reader, The hive columns num may greater than orc/parquet file columns num. So I think we need make orc/parquet file source support user-defined schema too.

Reply:
When the hive connector upgrade is completed, this piece will be changed, what do you think?